doc/tutorials/platformer/step_1.md

```diff
@@ -0,0 +1,99 @@
+# 1. Preparation
+
+Before you begin any kind of game project, you need an idea of what you want to make and I like to
+then give it a **name**. For this tutorial and game, I got inspiration from @spydon's tweet about
+the Flame game engine GitHub repository reaching 7000 stars and I thought, excellent! Ember will be
+on a quest to gather as many stars as possible and I will call the game, `Ember Quest`.
+
+![Tweet](../../images/tutorials/platformer/tweet.jpg)
+
+Now it is time to get started but first, you need to go to the [bare flame game
+tutorial](../bare_flame_game.md) and complete the necessary setup steps. When you come back, you
+should already have the `main.dart` file with the following content:
+
+```dart
+import 'package:flame/game.dart';
+import 'package:flutter/widgets.dart';
+
+void main() {
+  final game = FlameGame();
+  runApp(GameWidget(game: game));
+}
+```
+
+
+## Planning
+
+Like in the [klondike](../klondike/klondike.md) tutorial, starting a new game can feel overwhelming.
+I like to first decide what platform I am trying to target. Will this be a mobile game, a desktop
+game, or maybe a web game, with Flutter and Flame, these are all possible.  For this game though, I
+am going to focus on a web game.  This means my users will interact with the game using their
+keyboards.  
+
+Starting with a simple sketch (it doesn't have to be perfect as mine is very rough) is
+the best way to get an understanding of what will need to be accomplished. Take for instance the
+below sketch, we know we will need the following:
+
+- Player Class
+- Enemy Class
+- Star Class
+- Platform Class
+- Ground Class
+- HUD Class (health and stars collected)
+
+![Sketch of Ember Quest](../../images/tutorials/platformer/ember_quest_sketch.png)
+
+All of these will be brought together in `EmberQuestGame` derived from `FlameGame`.
+
+
+## Assets
+
+Every game needs assets.  Assets are images, sprites, animations, sounds, etc. Now, I am not an
+artist, but because I am basing this game on Ember, the flame mascot, and Ember is already designed,
+it sets the tone that this will be a pixel art game.  There are numerous sites available that
+provide free pixel art that can be used in games, but please check and comply with the licensing and
+always provide valid creator attribution.  For this game though, I am going to take a chance and
+make my artwork using an online pixel art tool.  If you decide to use this tool, multiple online
+tutorials will assist you with the basic operations as well as exporting the assets.  Now normally,
+most games will utilize spritesheets.  These combine many images into one larger image that can be
+sectioned and used as individual images.  For this tutorial though, I specifically will save the
+images individually as I want to demonstrate the Flame engines' caching abilities.  Ember and the
+water enemy are spritesheets though as they contain multiple images to create animations.
```

Comment:
```suggestion
Every game needs assets. Assets are images, sprites, animations, sounds, etc. Now, I am not an
artist, but because I am basing this game on Ember, the flame mascot, and Ember is already designed,
it sets the tone that this will be a pixel art game. There are numerous sites available that
provide free pixel art that can be used in games, but please check and comply with the licensing and
always provide valid creator attribution. For this game though, I am going to take a chance and
make my artwork using an online pixel art tool. If you decide to use this tool, multiple online
tutorials will assist you with the basic operations as well as exporting the assets. Now normally,
most games will utilize spritesheets. These combine many images into one larger image that can be
sectioned and used as individual images. For this tutorial though, I specifically will save the
images individually as I want to demonstrate the Flame engines' caching abilities. Ember and the
water enemy are spritesheets though as they contain multiple images to create animations.
```
For some strange reason there are double spaces everywhere?

Reply:
More than likely I fell into my old way of writing.  All through school and uni APA - 2 space was a requirement and its an old habit to break. https://www.instructionalsolutions.com/blog/one-space-vs-two-after-period